config-generator/src/main/groovy/org/infinispan/images/ConfigGenerator.groovy

```diff
@@ -98,9 +96,13 @@ defaultConfig.jgroups.bindAddress = InetAddress.localHost.hostAddress
 // If no user config then use defaults, otherwise load user config and add default values for missing elements
 Map configYaml = args.length == 2 ? defaultConfig : mergeMaps(defaultConfig, new Yaml().load(new File(args[2]).newInputStream()))
 
-// Create Keystore if required
-createKeystore configYaml.keystore, outputDir
-
+// Create Keystore if required, else ctrPass can be used to
+// pass the output dir to the xml generator
+// it may be used by Infinispan to store the generate-self-signed-certificate
+if (configYaml.keystore.crtPath != null) createKeystore configYaml.keystore, outputDir
+    else if ( configYaml.keystore.selfSignCert) {
+     configYaml.keystore.crtPath = outputDir
+    }
```

Comment:
I think this logic should be in the `createKeystore` method, but we should update the name to `configureKeystore`. That way we can just do `ks.` and the logic is in one place.